### ISSUE TITLE
Feature request: Download your data

### ISSUE THREAD

**Issue by mdrenger** [hackmdio/hackmd#823]
i'd like to see a 'download your data'-function in hackmd similar to google takeout.

reasons:
- it's a nice thing to have
legal: 
offering hackmd-services to europeans comes with legal responsibilities (general data protection regulation GDPR)
- offering users the possibility of backups and admins an export function ( right of access, Art. 15 GDPR)
- possible data portability by uploading a  backup to another hackmd-instance ( data portability, Art 20 GDPR)

**Comment by ccoenen** [comment]
Yes, this is a very good suggestion. I don't believe anyone is working on it at the moment.

There's also an interesting question involved: what are "your" documents? everything that you "own" (that you created), or everything you ever worked on? Perhaps even everything you've ever looked at?

GDPR wasn't exactly written with a service like HackMD in mind, so this is probably debateable. But I think at least 1 and 2 should be in an export.

**Comment by ccoenen** [comment]
This is somewhat related to the whole missing user section, which is tracked in #314 and #653 goes in a somewhat similar direction (although is not intended for downloading)

**Comment by SISheogorath** [comment]
I already added a download option for all notes the user owns:

https://github.com/hackmdio/hackmd/pull/830

Right now without the ability to download the full history, due to some technical issues I may find time to tackle another day.

But you can delete you account now and export data. As soon as 1.2.0 is release this will make to all up-to-date HackMD or more correct future CodiMD instances.
